File: 1790-Check-if-One-String-Swap-Can-Make-Strings-Equal/main.py
class Solution:
    def areAlmostEqual(self, s1: str, s2: str) -> bool:
        
        if s1 == s2:
            return True

        diff_count = 0
        diff_indices = []
        for i in range(len(s1)):
            if s1[i] != s2[i]:
                diff_count += 1
                diff_indices.append(i)
                
                if diff_count > 2:
                    return False

        return diff_count == 2 and s1[diff_indices[0]] == s2[diff_indices[1]] and s1[diff_indices[1]] == s2[diff_indices[0]]

File: 1790-Check-if-One-String-Swap-Can-Make-Strings-Equal/main.go
package main

func areAlmostEqual(s1 string, s2 string) bool {

	if s1 == s2 {
		return true
	}

	diffCount := 0
	var diffIndeces []int

	for i := 0; i < len(s1); i++ {
		if s1[i] != s2[i] {
			diffCount += 1
			diffIndeces = append(diffIndeces, i)

			if diffCount > 2 {
				return false
			}
		}
	}

	return diffCount == 2 && s1[diffIndeces[0]] == s2[diffIndeces[1]] && s1[diffIndeces[1]] == s2[diffIndeces[0]]

}
